core/services/hash_utils.py

```
@@ -0,0 +1,78 @@
+"""
+SMS Bridge v2.2 - Hash Generation Utility
+Generates Base32-encoded HMAC-SHA256 hashes for onboarding.
+"""
+import base64
+import hashlib
+import hmac
+import secrets
+from datetime import datetime
+from typing import Tuple
```

Ruff (0.14.10)

`typing.Tuple` is deprecated, use `tuple` instead

(UP035)